packages/nextjs/src/config/webpack.ts

```diff
@@ -82,13 +82,22 @@ export function constructWebpackConfigFunction(
       if (userSentryOptions.autoInstrumentServerFunctions !== false) {
         const pagesDir = newConfig.resolve?.alias?.['private-next-pages'] as string;
 
+        // Next.js allows users to have the `pages` folder inside a `src` folder, however "`src/pages` will be ignored
+        // if `pages` is present in the root directory"
+        // - https://nextjs.org/docs/advanced-features/src-directory
+        const shouldIncludeSrcDirectory = !fs.existsSync(path.resolve(projectDir, 'pages'));
+
         // Default page extensions per https://github.com/vercel/next.js/blob/f1dbc9260d48c7995f6c52f8fbcc65f08e627992/packages/next/server/config-shared.ts#L161
         const pageExtensions = userNextConfig.pageExtensions || ['tsx', 'ts', 'jsx', 'js'];
         const pageExtensionRegex = pageExtensions.map(escapeStringForRegex).join('|');
 
         newConfig.module.rules.push({
           // Nextjs allows the `pages` folder to optionally live inside a `src` folder
-          test: new RegExp(`${escapeStringForRegex(projectDir)}(/src)?/pages/.*\\.(${pageExtensionRegex})`),
+          test: new RegExp(
+            `${escapeStringForRegex(projectDir)}${
+              shouldIncludeSrcDirectory ? '(/src)?' : ''
```

Comment:
l: Do we need the `?` on `(/src)?` now?

Reply:
Nope, don't need it anymore. Thanks! 652e137611015717cd391894876a87ba941d81b3